src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,87 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+type Props = {
+  selectedUserId: number;
+  selectedTodo: Todo | null;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User>(defaultUser);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(setUser)
+      .catch(() => console.error('Error while trying to load user!'));
+  }, [selectedUserId]);
+
+  const isLoaded = selectedTodo && user !== defaultUser;
```

Comment:
The condition `user !== defaultUser` is used to determine if the user data is loaded. This might not be the most reliable way to check if the user data is loaded, as it relies on the default user object. Consider using a separate state variable to track the loading status of the user data.